src/components/application_manager/test/commands/mobile/alert_request_test.cc

```diff
@@ -0,0 +1,237 @@
+};
+
+class CallOnTimeOut {
+ public:
+  CallOnTimeOut(CommandRequestImpl& command) : command_(command) {}
+
+  void operator()() {
+    command_.onTimeOut();
+  }
+
+  CommandRequestImpl& command_;
+};
+
+TEST_F(AlertRequestTest, OnTimeOut_UNSUCCESS) {
+  MessageSharedPtr msg(CreateMessage(smart_objects::SmartType_Map));
+  (*msg)[am::strings::msg_params][am::strings::soft_buttons] = 0;
+  CommandPtr command(CreateCommand<AlertRequest>(msg));
+
+  command->onTimeOut();
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).Times(0);
+}
+
+TEST_F(AlertRequestTest, OnTimeOut_SUCCESS) {
+  CommandPtr command(CreateCommand<AlertRequest>());
+
+  MessageSharedPtr sended_result_msg(
+      CreateMessage(smart_objects::SmartType_Null));
+  EXPECT_CALL(
+      *MockMessageHelper::message_helper_mock(),
+      CreateNegativeResponse(_, _, _, mobile_apis::Result::GENERIC_ERROR))
+      .WillOnce(Return(sended_result_msg));
+
+  MessageSharedPtr received_result_msg(
+      CatchMobileCommandResult(CallOnTimeOut(*command)));
+  EXPECT_EQ(sended_result_msg, received_result_msg);
+}
+
+TEST_F(AlertRequestTest, Run_ApplicationIsNotRegistered_UNSUCCESS) {
+  CommandPtr command(CreateCommand<AlertRequest>());
+
+  ON_CALL(app_mngr_, application(_))
+      .WillByDefault(Return(ApplicationSharedPtr()));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::APPLICATION_NOT_REGISTERED,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertRequestTest, Run_AlertFrequencyIsTooHigh_UNSUCCESS) {
+  CommandPtr command(CreateCommand<AlertRequest>());
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+
+  ON_CALL(*app, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+  ON_CALL(*app, AreCommandLimitsExceeded(_, _)).WillByDefault(Return(true));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::REJECTED,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertRequestTest, Run_FailToProcessSoftButtons_UNSUCCESS) {
+  const mobile_apis::Result::eType kResultCode =
+      mobile_apis::Result::INVALID_ENUM;
+
+  CommandPtr command(CreateCommand<AlertRequest>());
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+  ON_CALL(*app, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+  ON_CALL(*app, AreCommandLimitsExceeded(_, _)).WillByDefault(Return(false));
+
+  ON_CALL(app_mngr_, GetPolicyHandler())
+      .WillByDefault(
+          ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
```

Comment:
@okozlovlux A NULL pointer points to regions filled with dragons, demons, core dumps, and numberless other foul creatures, all of which delight in frolicing in thy program if thou disturb their sleep. [The Ten Commandments for C Programmers](https://www.lysator.liu.se/c/ten-commandments.html)